crypto/src/main/java/org/web3j/crypto/ECKeyPair.java

```diff
@@ -39,13 +41,18 @@ public BigInteger getPublicKey() {
      * @return  An {@link ECDSASignature} of the hash
      */
     public ECDSASignature sign(byte[] transactionHash) {
+        return Async.sync(signAsync(transactionHash));
+    }
+
+    public CompletableFuture<ECDSASignature> signAsync(byte[] transactionHash) {
```

Comment:
To my point above, can't we leave the default sync as is, then just wrap the async versions in a completable future. I'm not a fan of doing this all async first, as it results in lots of futures embedded in the methods which is unnecessary IMHO.

Reply:
That's easy enough to do (I will do that) in this case at the bottom of the stack. Where it gets more interesting is in `Sign.java`. You have a bunch of computation depending on a result that _could_ be delivered async. None of the functionality in that class is packaged up in such a way that someone can reuse it if they were to want async execution.

I just want to check something: you don't like this for reasons other than perf right? Because perf wise I am sure this introduces negligible overhead. It's just going to be a couple of CAS ops / extra objects. I'd be happy to provide jmh benches.

Alternative way forward: I do the wrapping as suggested (sync code stays the same, extra method, wraps sync call in completed future), _but_ inside of `Sign` I make a couple of protected methods that make it easy for someone extending to repackage that code to use an inner async signature call.

Thanks for sticking with me!